pkg/certificate/manager_test.go

```diff
@@ -68,14 +56,55 @@ var _ = Describe("Test Certificate Manager", func() {
 	})
 })
 
+var _ = Describe("Test Rotor", func() {
+
+	cn := CommonName("foo")
+
+	Context("Testing rotating expiring certificates", func() {
+
+		validityPeriod := -1 * time.Hour // negative time means this cert has already expired -- will be rotated asap
+
+		stop := make(chan struct{})
+		defer close(stop)
+		msgBroker := messaging.NewBroker(stop)
+		certManager, err := NewManager(&Certificate{}, &fakeIssuer{}, validityPeriod, msgBroker, make(chan struct{}))
```

Comment:
You could reuse the channel created earlier in this test to stop the message broker and cert rotation at the same time, unless it's important their lifetimes should be different (which I have a feeling isn't the case):
```suggestion
		certManager, err := NewManager(&Certificate{}, &fakeIssuer{}, validityPeriod, msgBroker, stop)
```